tests/test_maintel_m1m3_check_actuators.py

```diff
@@ -0,0 +1,128 @@
+            side_effect=self.mock_test_bump
+        )
+        self.script.mtcs.enter_m1m3_engineering_mode = unittest.mock.AsyncMock()
+        self.script.mtcs.exit_m1m3_engineering_mode = unittest.mock.AsyncMock()
+        self.script.mtcs.assert_liveliness = unittest.mock.AsyncMock()
+        self.script.mtcs.assert_all_enabled = unittest.mock.AsyncMock()
+        self.script.mtcs.assert_m1m3_detailed_state = unittest.mock.AsyncMock()
+        self.script.mtcs.assert_feasibility = unittest.mock.AsyncMock()
+
+        self.bump_test_status = types.SimpleNamespace(
+            testState=[BumpTest.NOTTESTED] * len(self.m1m3_actuator_ids)
+        )
+
+        return (self.script,)
+
+    async def mock_test_bump(self, actuator_id):
+        await asyncio.sleep(5.0)
+        self.bump_test_status.testState[actuator_id] = BumpTest.PASSED
+
+    async def test_configure_good(self):
+        """Testing a valid configuration"""
+
+        # Configure with "all" actuators
+        async with self.make_script():
+            actuators = "all"
+
+            await self.configure_script(
+                actuators=actuators
+            )
+
+            assert self.script.actuators == self.m1m3_actuator_ids
+
+        # Try configure with a list of valid actuators ids
+        async with self.make_script():
+            actuators = [101, 210, 301, 410]
+
+            await self.configure_script(
+                actuators=actuators,
+            )
+
+            assert self.script.actuators == actuators
+
+    async def test_configure_bad(self):
+        """Testing an invalid configuration"""
+
+        async with self.make_script():
+            # Invalid actuators: 501 and 505
+            actuators = [101, 210, 301, 410, 501, 505]
+
+            # Check if actuators are valid
+            # if not all([actuator in self.m1m3_actuator_ids for actuator
+            # in actuators]):
+            #     raise ValueError("Invalid actuator ID.")
+
+            # If actuators_bad_ids is not empty, it should raise a ValueError
+            actuators_bad_ids = [
+                actuator
+                for actuator in actuators
+                if actuator not in self.m1m3_actuator_ids
+            ]
+            if actuators_bad_ids:
+                with self.assertRaises(ValueError):
+                    await self.configure_script(
+                        actuators=actuators_bad_ids,
+                    )
+
+    async def test_run(self):
+        # Run the script
+        async with self.make_script():
+            # Assert feasibility
+            self.script.mtcs.assert_feasibility()
+
+            actuators = "all"
+            await self.configure_script(actuators=actuators)
+
+            await self.run_script()
+
+            assert all(
+                [
+                    self.script.bump_test_status.testState[actuator] == BumpTest.PASSED
+                    for actuator in self.script.actuators
+                ]
+            )
+
```

Comment:
I think you could do some additional assertions about the expected calls done in the method you mocked.

For example, you expect `self.script.mtcs.enter_m1m3_engineering_mode` to be awaited once, so you can add

```
self.script.mtcs.enter_m1m3_engineering_mode.assert_awaited_once()
```

and the same for all the other methods that need to be awaited once. 

Also, when you run with `actuators="all"` you expect `self.script.mtcs.run_m1m3_actuator_bump_test` to be awaited with all the hp `actuator_id` and the appropriate `primary` and `secondary` values. 

you can do something like this:

```
expected_calls = [
    unittest.mock.call(
        actuator_id=actuator_id,
        primary=True,
        secondary=self.script.has_secondary_actuator(actuator_id)
    ) for actuator_id in self.script.m1m3_actuator_ids
]

self.script.mtcs.run_m1m3_actuator_bump_test.assert_has_calls(expected_calls)
```

Reply:
`self.script.mtcs.assert_feasibility` doesn't exist and you should not mock `self.script.assert_feasibility`